lib/tree.rb

```diff
@@ -16,42 +16,76 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def add(key, value)
-    raise NotImplementedError
+  # Time Complexity: O(log(n))
+  # Space Complexity: O(log(n))
+  def add(key, value = nil, current = @root)
```

Comment:
👍 Good use of default parameters.  The time/space complexities are correct if the tree is balanced and O(n) otherwise.